tests/composition/test_autodiffcomposition.py

```diff
@@ -154,7 +154,7 @@ def test_xor_training_correctness(self, eps, calls, opt, autodiff_mode, benchmar
 
         assert len(results) == len(expected)
         for r, t in zip(results, expected):
```

Comment:
Can the above assert len, and the zip iteration be replaced by a single `np.testing.assert_allclose`?

Reply:
Fixing